extension/src/popup/components/mnemonicPhrase/MnemonicDisplay/index.tsx

```diff
@@ -1,8 +1,50 @@
 import React from "react";
 import { Card } from "@stellar/design-system";
+// @ts-ignore
+import { generateMnemonic } from "stellar-hd-wallet";
+import { random } from "lodash";
 
 import "./styles.scss";
 
+interface generateMnemonicPhraseDisplayProps {
+  mnemonicPhrase: string;
+}
+
+export const generateMnemonicPhraseDisplay = ({
+  mnemonicPhrase,
+}: generateMnemonicPhraseDisplayProps) =>
+  mnemonicPhrase.split(" ").map((word: string) => {
+    const randomNumber = random(1, 10);
+    const randomWordArr = generateMnemonic().split(" ");
+    const randomWordIndex = random(0, randomWordArr.length);
+    const randomWord = randomWordArr[randomWordIndex];
```

Comment:
when we show the mnemonic phrase, we're going to add some hidden, random words throughout the phrase to obscure the actual words: https://halborn.com/disclosures/demonic-vulnerability/